src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -964,13 +952,263 @@ def load_module(name, value):
+                ignore_patterns = ["*.bin", "*.msgpack"]
+
+                safetensors_variant_filenames = set([f for f in variant_filenames if f.endswith(".safetensors")])
+                safetensors_model_filenames = set([f for f in model_filenames if f.endswith(".safetensors")])
+                if (
+                    len(safetensors_variant_filenames) > 0
+                    and safetensors_model_filenames != safetensors_variant_filenames
+                ):
+                    logger.warn(
+                        f"\nA mixture of {variant} and non-{variant} filenames will be loaded.\nLoaded {variant} filenames:\n[{', '.join(safetensors_variant_filenames)}]\nLoaded non-{variant} filenames:\n[{', '.join(safetensors_model_filenames - safetensors_variant_filenames)}\nIf this behavior is not expected, please check your folder structure."
+                    )
+            else:
+                ignore_patterns = ["*.safetensors", "*.msgpack"]
+
+                bin_variant_filenames = set([f for f in variant_filenames if f.endswith(".bin")])
+                bin_model_filenames = set([f for f in model_filenames if f.endswith(".bin")])
+                if len(bin_variant_filenames) > 0 and bin_model_filenames != bin_variant_filenames:
+                    logger.warn(
+                        f"\nA mixture of {variant} and non-{variant} filenames will be loaded.\nLoaded {variant} filenames:\n[{', '.join(bin_variant_filenames)}]\nLoaded non-{variant} filenames:\n[{', '.join(bin_model_filenames - bin_variant_filenames)}\nIf this behavior is not expected, please check your folder structure."
+                    )
+
+            if config_is_cached:
+                re_ignore_pattern = [re.compile(fnmatch.translate(p)) for p in ignore_patterns]
+                re_allow_pattern = [re.compile(fnmatch.translate(p)) for p in allow_patterns]
+
+                expected_files = [f for f in filenames if not any(p.match(f) for p in re_ignore_pattern)]
+                expected_files = [f for f in expected_files if any(p.match(f) for p in re_allow_pattern)]
+
+                folder_name = f"models--{'--'.join(pretrained_model_name_or_path.split('/'))}"
+                cached_pipeline = os.path.join(cache_dir, folder_name, "snapshots", commit_hash)
+
+                pipeline_is_cached = all(os.path.isfile(os.path.join(cached_pipeline, f)) for f in expected_files)
+
+                if pipeline_is_cached:
+                    # if the pipeline is cached, we can directly return it
+                    # else call snapshot_download
+                    return cached_pipeline
+
+        # download all allow_patterns - ignore_patterns
+        cached_folder = snapshot_download(
+            pretrained_model_name_or_path,
+            cache_dir=cache_dir,
+            resume_download=resume_download,
+            proxies=proxies,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            revision=revision,
```

Comment:
You can use the `commit_hash` here since you already know it

```suggestion
            revision=commit_hash,
```

Reply:
If `local_files_only` we still get an ugly "File does not exist" error if the pipeline is not cached, instead of a user-friendly error message. I believe this is because this:
https://github.com/huggingface/huggingface_hub/blame/main/src/huggingface_hub/_snapshot_download.py#L166 
happens before this:
https://github.com/huggingface/huggingface_hub/blame/main/src/huggingface_hub/_snapshot_download.py#LL174
Maybe we should catch that exception in `hfh`?

Reply:
cc @Wauplin